File: backend/fief/schemas/email_template.py
from fief.schemas.generics import BaseModel, CreatedUpdatedAt, UUIDSchema
from fief.services.email_template.types import EmailTemplateType


class EmailTemplateUpdate(BaseModel):
    subject: str | None
    content: str | None


class EmailTemplateBase(UUIDSchema, CreatedUpdatedAt):
    type: EmailTemplateType
    subject: str
    content: str


class EmailTemplate(EmailTemplateBase):
    pass

File: admin-frontend/src/schemas/email-template.ts
import { CreatedUpdatedAt, UUIDSchema } from './generics';

export enum EmailTemplateType {
  BASE = 'BASE',
  WELCOME = 'WELCOME',
  FORGOT_PASSWORD = 'FORGOT_PASSWORD',
}

export interface EmailTemplateUpdate {
  subject?: string;
  content?: string;
}

interface EmailTemplateBase extends UUIDSchema, CreatedUpdatedAt {
  type: EmailTemplateType;
  subject: string;
  content: string;
}

export interface EmailTemplate extends EmailTemplateBase {}

export interface EmailTemplatePreview {
  subject: string;
  content: string;
}
